### PR TITLE
A client for Nebula Graph Database that support multi versions

### PR DESCRIPTION
**Pull request by veezhang**
A client for `Nebula Graph Database` that support multi versions.

Those files in `ccore/nebula/internal/thrift` is copy from https://github.com/vesoft-inc/nebula-go .

### CODE REVIEW

**Review by nianiaJR — APPROVED** [review]
LGTM

File: ccore/nebula/options.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		o.storageAdmin.tlsConfig = tlsConfig
+	}
+}
+
+func (o *Options) complete() {
+	defaultOpts := defaultOptions()
+
+	if o.log == nil {
+		o.log = defaultOpts.log
+	}
```
Comment (kqzh) [inline ccore/nebula/options.go:155]: why don't use `if o.log == nil {
o.log = noOpLogger{}
}` instead, don't need define defaultOpts

and check o.version same as o.log?

File: ccore/nebula/alias.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	Version           = types.Version
+	AuthResponse      = types.AuthResponse
+	ExecutionResponse = types.ExecutionResponse
+)
+
+var (
+	V2_0_0 = types.V2_0_0
+	V2_5_0 = types.V2_5_0
+	V2_5_1 = types.V2_5_1
+	V2_6_0 = types.V2_6_0
```
Comment (kqzh) [inline ccore/nebula/alias.go:17]: https://github.com/vesoft-inc/nebula-go/pull/148/files, I see thrift code use 2.6.0, so we need fix this compile bug too, or we use 2.6.1 thrift code instead

Comment (veezhang) [reply ccore/nebula/alias.go:17]: I have fixed.

File: ccore/nebula/options.go (hunk 1 above)
Comment (veezhang) [reply ccore/nebula/options.go:155]: Use the same entrance `defaultOptions`. If this future changes, only `defaultOptions` need to chanegd.

File: ccore/nebula/client.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import "github.com/vesoft-inc/nebula-http-gateway/ccore/nebula/types"
+
+type (
+	// TODO: add client pool management
+
+	Client interface {
+		Graph() GraphClient
+		Meta() MetaClient
+		StorageAdmin() StorageAdminClient
```
[the remark this answers is not in the document]
Comment (veezhang) [reply ccore/nebula/client.go:11]: This is named according to `thrift` define, it is best not to introduce new terms.

File: ccore/nebula/client_meta.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		meta:         newDriverMeta(endpoints, &o.meta),
+	}, nil
+}
+
+func (c *defaultMetaClient) Open() error {
+	return c.meta.open(c.driver)
+}
+
+func (c *defaultMetaClient) Close() error {
+	return c.meta.open(c.driver)
```
Comment (kqzh) [inline ccore/nebula/client_meta.go:41]: use `c.meta.close()`?

File: ccore/nebula/client_graph.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+func (c *defaultGraphClient) ExecuteJson(stmt []byte) ([]byte, error) {
+	if err := c.graph.open(c.driver); err != nil {
+		return nil, err
+	}
+
+	return c.graph.ExecuteJson(c.graph.sessionId, stmt)
+}
+
+func (c *defaultGraphClient) Close() error {
+	return c.graph.open(c.driver)
```
Comment (kqzh) [inline ccore/nebula/client_graph.go:55]: use `c.graph.close()` ?

File: ccore/nebula/client_storage_admin.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+		storageAdmin: newDriverStorageAdmin(endpoints, &o.storageAdmin),
+	}, nil
+}
+
+func (c *defaultStorageAdminClient) Open() error {
+	return c.storageAdmin.open(c.driver)
+}
+
+func (c *defaultStorageAdminClient) Close() error {
+	return c.storageAdmin.open(c.driver)
```
Comment (kqzh) [inline ccore/nebula/client_storage_admin.go:41]: use `c.storageAdmin.close()` ?

**Review by kqzh — APPROVED** [review]
LGTM

**Pull request merged**